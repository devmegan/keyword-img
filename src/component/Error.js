export default function Error({ error, setError }) {

    const handleClick = () => {
        setError(false);
    }

    return (
        <div className="alert alert-error shadow-lg max-w-sm mx-auto">
            <div>
                <svg 
                    className="stroke-current flex-shrink-0 h-6 w-6 hover:text-red-800 cursor-pointer" 
                    onClick={handleClick}
                    fill="none" 
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                >
                    <path 
                        d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" 
                        strokeLinecap="round" 
                        strokeLinejoin="round" 
                        strokeWidth="2" 
                    />
                </svg>
                <span>{error}</span>
            </div>
        </div>
    )
}
