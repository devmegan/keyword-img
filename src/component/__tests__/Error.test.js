import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Error from '../Error';

const errorFn = jest.fn();

describe('error', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('renders w/o crashing', () => {
    expect.hasAssertions();

    const view = render(<Error />);
    
    expect(view).toMatchSnapshot();
  });

  it('hides error on click', () => {
    expect.hasAssertions();

    render(<Error setError={errorFn}/>);

    userEvent.click(screen.getByRole('button'));

    expect(screen.getByRole('alert')).toBeInTheDocument();
    expect(errorFn).toHaveBeenCalledWith(false);
  });
});