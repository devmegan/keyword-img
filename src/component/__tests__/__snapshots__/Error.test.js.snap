// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error renders w/o crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert alert-error shadow-lg max-w-sm mx-auto"
        role="alert"
      >
        <div>
          <svg
            class="stroke-current flex-shrink-0 h-6 w-6 hover:text-red-800 cursor-pointer"
            fill="none"
            role="button"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert alert-error shadow-lg max-w-sm mx-auto"
      role="alert"
    >
      <div>
        <svg
          class="stroke-current flex-shrink-0 h-6 w-6 hover:text-red-800 cursor-pointer"
          fill="none"
          role="button"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <span />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
